src/lib/stepper/stepper.spec.ts

```diff
@@ -0,0 +1,559 @@
+  nextButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(1);
+
+  nextButtonNativeEl = fixture.debugElement
+      .queryAll(By.directive(MdStepperNext))[1].nativeElement;
+  nextButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(2);
+
+  nextButtonNativeEl = fixture.debugElement
+      .queryAll(By.directive(MdStepperNext))[2].nativeElement;
+  nextButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(2);
+}
+
+function checkPreviousStepperButton(stepperComponent: MdHorizontalStepper | MdVerticalStepper,
+                                    fixture: ComponentFixture<any>) {
+  expect(stepperComponent.selectedIndex).toBe(0);
+
+  stepperComponent.selectedIndex = 2;
+  let previousButtonNativeEl = fixture.debugElement
+      .queryAll(By.directive(MdStepperPrevious))[2].nativeElement;
+  previousButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(1);
+
+  previousButtonNativeEl = fixture.debugElement
+      .queryAll(By.directive(MdStepperPrevious))[1].nativeElement;
+  previousButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(0);
+
+  previousButtonNativeEl = fixture.debugElement
+      .queryAll(By.directive(MdStepperPrevious))[0].nativeElement;
+  previousButtonNativeEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(0);
+}
+
+function checkStepPosition(stepperComponent: MdHorizontalStepper | MdVerticalStepper,
+                           fixture: ComponentFixture<any>) {
+  expect(stepperComponent._getAnimationDirection(0)).toBe('current');
+  expect(stepperComponent._getAnimationDirection(1)).toBe('next');
+  expect(stepperComponent._getAnimationDirection(2)).toBe('next');
+
+  stepperComponent.selectedIndex = 1;
+  fixture.detectChanges();
+
+  expect(stepperComponent._getAnimationDirection(0)).toBe('previous');
+  expect(stepperComponent._getAnimationDirection(1)).toBe('current');
+  expect(stepperComponent._getAnimationDirection(2)).toBe('next');
+
+  stepperComponent.selectedIndex = 2;
+  fixture.detectChanges();
+
+  expect(stepperComponent._getAnimationDirection(0)).toBe('previous');
+  expect(stepperComponent._getAnimationDirection(1)).toBe('previous');
+  expect(stepperComponent._getAnimationDirection(2)).toBe('current');
+}
+
+function checkKeyboardEvent(stepperComponent: MdHorizontalStepper | MdVerticalStepper,
+                            fixture: ComponentFixture<any>,
+                            stepHeader: DebugElement[]) {
+  expect(stepperComponent._focusIndex).toBe(0);
+  expect(stepperComponent.selectedIndex).toBe(0);
+
+  let stepHeaderEl = stepHeader[0].nativeElement;
+  dispatchKeyboardEvent(stepHeaderEl, 'keydown', RIGHT_ARROW);
+  fixture.detectChanges();
+
+  expect(stepperComponent._focusIndex).toBe(1);
```

Comment:
I think fallback messages for these would be helpful to guide us through your thinking.